packages/api-react/src/hooks/usePrefs.ts

```diff
@@ -0,0 +1,56 @@
+import { useCallback, useEffect, useState } from 'react';
+import EventEmitter from '../utils/EventEmitter';
+
+const eventEmitter = new EventEmitter();
+
+function deepEqual(objA: any, objB: any){
+  return JSON.stringify(objA) === JSON.stringify(objB);
+}
+
+export default function usePrefs<T>(
```

Comment:
Can you add a type constraint on `T` to ensure that it's a JSON-compatible type? Something like what's mentioned here seems reasonable, but I'm open to other suggestions: https://dev.to/_gdelgado/type-constraints-in-typescript-34ek